core/src/main/java/com/linecorp/armeria/internal/common/Http1ObjectEncoder.java

```diff
@@ -112,13 +116,45 @@ protected final ChannelFuture writeNonInformationalHeaders(int id, HttpObject co
         } else {
             f = write(id, converted, false);
             if (endStream) {
-                f = write(id, LastHttpContent.EMPTY_LAST_CONTENT, true);
+                final ChannelPromise promise = channel().newPromise();
+                combine(promise, f, write(id, LastHttpContent.EMPTY_LAST_CONTENT, true));
+                f = promise;
```

Comment:
We could make `combine()` return a new promise:

```suggestion
                f = combine(f, write(id, LastHttpContent.EMPTY_LAST_CONTENT, true));
```

.. so it looks more readable.